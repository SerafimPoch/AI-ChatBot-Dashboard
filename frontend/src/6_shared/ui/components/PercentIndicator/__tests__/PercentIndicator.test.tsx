import React from "react";
import { render } from "@testing-library/react";
import {
  PercentIndicatorContainer,
  PercentIndicatorTriangle,
  PercentIndicatorValue,
} from "../PercentIndicator.styles";

describe("PercentIndicator page components", () => {
  it("PercentIndicatorContainer renders correctly", () => {
    const { asFragment } = render(<PercentIndicatorContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("PercentIndicatorTriangle renders correctly", () => {
    const { asFragment } = render(
      <PercentIndicatorTriangle isPercentNegative />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("PercentIndicatorValue renders correctly", () => {
    const { asFragment } = render(<PercentIndicatorValue isPercentNegative />);
    expect(asFragment()).toMatchSnapshot();
  });
});
