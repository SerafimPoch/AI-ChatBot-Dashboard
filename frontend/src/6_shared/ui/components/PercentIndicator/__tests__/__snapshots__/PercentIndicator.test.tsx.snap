// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentIndicator page components PercentIndicatorContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`PercentIndicator page components PercentIndicatorTriangle renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom-style: solid;
  border-bottom-width: 10px;
  border-bottom-color: #EA7465;
  rotate: 180deg;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`PercentIndicator page components PercentIndicatorValue renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  color: #EA7465;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;
